libs/counterfactuals/src/util/largeCounterfactualsView/getCounterfactualsScatter.ts

```diff
@@ -0,0 +1,224 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import {
+  FluentUIStyles,
+  IGenericChartProps,
+  JointDataset
+} from "@responsible-ai/core-ui";
+import { WhatIfConstants } from "@responsible-ai/interpret";
+import { localization } from "@responsible-ai/localization";
+import { PointMarkerOptionsObject } from "highcharts";
+import { Dictionary } from "lodash";
+
+import { buildScatterTemplate } from "./buildScatterTemplate";
+
+export interface IDatasetExplorerSeries {
+  name?: string;
+  color: any;
+  data: IDatasetExplorerData[];
+  marker?: PointMarkerOptionsObject;
+}
+export interface IDatasetExplorerData {
+  x: number;
+  y: number;
+  customData: any;
+  template: string | undefined;
+}
+
+export function getCounterfactualsScatter(
+  xSeries: number[],
+  ySeries: number[],
+  indexSeries: number[],
+  jointDataset: JointDataset,
+  selectedPointsIndexes: number[],
+  chartProps?: IGenericChartProps,
+  customPoints?: Array<{ [key: string]: any }>
+): any[] {
+  const dataSeries: any = [];
+  const result = [];
+  // const customData = plotlyProps.data[0].customdata;
+  const xData = xSeries;
```

Comment:
are these extra constants needed? Can we directly use xSeries?

Reply:
thank you, I might have added when I was doing logic on xSeries. Updated to remove these.